grails-app/services/com/redhat/theses/ThesisService.groovy

```diff
@@ -44,4 +44,24 @@ class ThesisService {
         }
         success
     }
+    
+    /**
+     * Transforms url into Set of Links
+     * 
+     * @param url - String which was returned from form
+     * @return Set with one Link
+     */
+    Set transformLinks(String url) {
```

Comment:
I think this should be a List, not a Set. You want to preserve the order because someone might want to put the most important link at the top and the less important link at the bottom.